l10n_ar_ux/models/account_payment.py

```diff
@@ -69,3 +69,9 @@ def action_post(self):
         # por esta razon es que tmb lo hacemos luego de super
         for rec in self.filtered(lambda x: x.l10n_latam_manual_checks and x.payment_method_line_id.code == 'check_printing'):
             rec._constrains_check_number_unique()
+
+    @api.constrains("currency_id", "company_id", "tax_withholding_id")
+    def _check_withholdings_and_currency(self):
+        for rec in self:
+            if rec.tax_withholding_id and rec.currency_id != rec.company_id.currency_id and rec.country_code == "AR":
+                raise UserError(_('Withholdings must be done in "%s" currency') % rec.company_id.currency_id.name)
```

Comment:
Constraint raise should always be ValidationError type
```suggestion
                raise ValidationError(_('Withholdings must be done in "%s" currency') % rec.company_id.currency_id.name)
```